### PR TITLE
Guidance: add index page with main navigation 

### PR DESCRIPTION
**Pull request by pmanrubia**
### Context

Ticket: https://dfedigital.atlassian.net/browse/CPDLP-2884

### Changes proposed in this pull request

1. Creates `index` page for the Guidance as per [figma designs](https://www.figma.com/file/Rz59oZYjWx6tod8zS4TFur/LPDOB-UCD-working-board?type=whiteboard&node-id=0-1&t=rGebkm7qmYs93i1N-0).
2. Automates pulling Release notes into the `index.page` including the `link to Github` to access the file.
3. Creates placeholders for the different sections of the main navigation
4. Add tests for the main navigation and for rendering the markdown page.

### Screenshot

image

### CODE REVIEW

File: app/helpers/admin_helper.rb [hunk 1]
```diff
@@ -9,4 +9,8 @@ def format_cohort(cohort)
   def admin_navigation_structure
     @admin_navigation_structure ||= NpqSeparation::NavigationStructures::AdminNavigationStructure.new
   end
+
+  def guidance_navigation_structure
+    @guidance_navigation_structure ||= NpqSeparation::NavigationStructures::GuidanceNavigationStructure.new
+  end
```
Comment (peteryates) [inline app/helpers/admin_helper.rb:15]: Should we have a `GuidanceHelper` for this? Or maybe a generic `NavigationHelper` where these methods can all live?

Doesn't feel quite right that the API guidance is in the admin helper.

File: app/views/api/guidance/index.html.erb [hunk 2]
```diff
@@ -1,11 +1,139 @@
...
+<div class="govuk-width-container">
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-full-from-desktop govuk-!-padding-top-8">
+      <div class="govuk-grid-column-one-third-from-desktop">
+        <h2 class="govuk-heading-l govuk-!-font-weight-bold">What you can do in the API</h2>
+        <p class="govuk-body">Guidance on how lead providers can integrate with this API so they can view, submit and
+          update data. The data enables providers to manage and deliver NPQ-based training, and ensure they get paid the
+          right amount by the DfE</p>
+        <p class="govuk-body govuk-!-margin-bottom-0">
+          <%= govuk_link_to "Browse guidance", api_guidance_path, class: "govuk-link govuk-!-font-weight-bold" %></p>
```
Comment (peteryates) [inline app/views/api/guidance/index.html.erb:55]: `govuk-link` is added by `govuk_link_to`

File: app/components/npq_separation/navigation_structures/guidance_navigation_structure.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+module NpqSeparation
```
Comment (cwrw) [inline app/components/npq_separation/navigation_structures/guidance_navigation_structure.rb:1]: could you add a spec for this, similar to `AdminNavigationStructure`?

File: app/views/layouts/api_guidance/_header.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<header class="govuk-header app-header" role="banner" data-module="govuk-header">
```
Comment (cwrw) [inline app/views/layouts/api_guidance/_header.html.erb:1]: could we use the following instead, and possibly not need the extra stylings for header?
```
<%=
  govuk_header(service_name: "NPQs data management API for lead providers") do |header|
  ...
  ```

**Review by cwrw — COMMENTED** [review]
looking good, some minor comments

File: app/models/release_notes.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+class ReleaseNotes
```
[the remark this answers is not in the document]
Comment (cwrw) [reply app/models/release_notes.rb:1]: hmmm fair enough, I was viewing it as something that accepts a file and returns the latest value, but makes sense

**Review by cwrw — APPROVED** [review]
nicely done! thanks @pmanrubia

File: app/views/api/guidance/index.html.erb [hunk 6]
```diff
@@ -1,11 +1,139 @@
...
+    <div class="govuk-grid-column-full-from-desktop">
+      <hr class="govuk-section-break govuk-section-break--xl govuk-section-break--visible">
+      <div class="govuk-grid-column-two-thirds ">
+        <h2 class="govuk-heading-l">API previous versions </h2>
+        <p class="govuk-body">
+          <%= govuk_link_to("View API v1.0.0", "#", class: "govuk-!-font-weight-bold") %>
+        </p>
+        <p class="govuk-body">
+          <%= govuk_link_to("View API v2.0.0", "#", class: "govuk-!-font-weight-bold") %>
+        </p>
```
Comment (peteryates) [inline app/views/api/guidance/index.html.erb:84]: I know this is just placeholder for now, but semantically this should be a list.

```suggestion
        <ul class="govuk-list">
          <li><%= govuk_link_to("View API v1.0.0", "#", class: "govuk-!-font-weight-bold") %></li>
          <li><%= govuk_link_to("View API v2.0.0", "#", class: "govuk-!-font-weight-bold") %></li>
        </ul>
```

File: app/views/api/guidance/index.html.erb [hunk 7]
```diff
@@ -1,11 +1,139 @@
...
+<div class="govuk-width-container">
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-full-from-desktop govuk-!-padding-top-8">
+      <div class="govuk-grid-column-one-third-from-desktop">
+        <h2 class="govuk-heading-l govuk-!-font-weight-bold">What you can do in the API</h2>
+        <p class="govuk-body">Guidance on how lead providers can integrate with this API so they can view, submit and
+          update data. The data enables providers to manage and deliver NPQ-based training, and ensure they get paid the
+          right amount by the DfE</p>
+        <p class="govuk-body govuk-!-margin-bottom-0">
+          <%= govuk_link_to "Browse guidance", api_guidance_path, class: "govuk-!-font-weight-bold" %></p>
```
Comment (peteryates) [inline app/views/api/guidance/index.html.erb:55]: Removing the bottom margin makes the text render a bit funny on mobile. I think just dropping the `govuk-!-margin-bottom-0` makes sense.

image

**Comment by peteryates** [comment]
Looking good, left some really minor comments about markup.

**Review by cwrw — APPROVED** [review]

**Pull request merged**